gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/CountStrategy.java

```diff
@@ -178,7 +178,7 @@ public void apply(final Traversal.Admin<?, ?> traversal) {
                             }
                         }
                     } else {
-                        TraversalHelper.insertBeforeStep(new RangeGlobalStep<>(traversal, 0L, highRange), curr, traversal);
+                        TraversalHelper.insertBeforeStep(new RangeGlobalStep<>(traversal, 0L, highRange < 0 ? 0 : highRange), curr, traversal);
```

Comment:
It looks like highRange is a Long, should these comparisons be in Long too?
```suggestion
                        TraversalHelper.insertBeforeStep(new RangeGlobalStep<>(traversal, 0L, highRange < 0L ? 0L : highRange), curr, traversal);
```

Reply:
`widening casting` should handle this

Reply:
Yes, this is more of a nit style/format comment since previous comparisons are done as Long.